pkg/api/types/image_types.go

```diff
@@ -0,0 +1,137 @@
+	Filters []string
+	// NameAndRefFilter filters images by name and reference
+	NameAndRefFilter []string
+	// Digests show digests (compatible with Docker, unlike ID)
+	Digests bool
+	// Names show image names
+	Names bool
+	// All (unimplemented yet, always true)
+	All bool
+}
+
+type ImageConvertCommandOptions struct {
+	GOptions GlobalCommandOptions
+
+	// Format the output using the given Go template, e.g, 'json'
+	Format string
+
+	// #region estargz flags
+	// Estargz convert legacy tar(.gz) layers to eStargz for lazy pulling. Should be used in conjunction with '--oci'
+	Estargz bool
+	// EstargzRecordIn read 'ctr-remote optimize --record-out=<FILE>' record file (EXPERIMENTAL)
+	EstargzRecordIn string
+	// EstargzCompressionLevel eStargz compression level
+	EstargzCompressionLevel int
+	// EstargzChunkSize eStargz chunk size
+	EstargzChunkSize int
+	// EstargzMinChunkSize the minimal number of bytes of data must be written in one gzip stream. (requires stargz-snapshotter >= v0.13.0)
+	EstargzMinChunkSize int
+	// EstargzExternalToc separate TOC JSON into another image (called "TOC image"). The name of TOC image is the original + "-esgztoc" suffix. Both eStargz and the TOC image should be pushed to the same registry. (requires stargz-snapshotter >= v0.13.0) (EXPERIMENTAL)
+	EstargzExternalToc bool
+	// EstargzKeepDiffID convert to esgz without changing diffID (cannot be used in conjunction with '--estargz-record-in'. must be specified with '--estargz-external-toc')
+	EstargzKeepDiffID bool
+	// #endregion
+
+	// #region zstd:chunked flags
+	// ZstdChunked convert legacy tar(.gz) layers to zstd:chunked for lazy pulling. Should be used in conjunction with '--oci'
+	ZstdChunked bool
+	// ZstdChunkedCompressionLevel zstd compression level
+	ZstdChunkedCompressionLevel int
+	// ZstdChunkedChunkSize zstd chunk size
+	ZstdChunkedChunkSize int
+	// ZstdChunkedRecordIn read 'ctr-remote optimize --record-out=<FILE>' record file (EXPERIMENTAL)
+	ZstdChunkedRecordIn string
+	// #endregion
+
+	// #region nydus flags
+	// Nydus convert legacy tar(.gz) layers to nydus for lazy pulling. Should be used in conjunction with '--oci'
+	Nydus bool
+	// NydusBuilderPath the nydus-image binary path, if unset, search in PATH environment
+	NydusBuilderPath string
+	// NydusWorkDir work directory path for image conversion, default is the nerdctl data root directory
+	NydusWorkDir string
+	// NydusPrefetchPatterns the file path pattern list want to prefetch
+	NydusPrefetchPatterns string
+	// NydusCompressor nydus blob compression algorithm, possible values: `none`, `lz4_block`, `zstd`, default is `lz4_block`
+	NydusCompressor string
+	// #endregion
+
+	// #region overlaybd flags
+	// Overlaybd convert tar.gz layers to overlaybd layers
+	Overlaybd bool
+	// OverlayFsType filesystem type for overlaybd
+	OverlayFsType string
+	// OverlaydbDBStr database config string for overlaybd
+	OverlaydbDBStr string
+	// #endregion
+
+	// #region generic flags
```

Comment:
nit: maybe we should put generic/common flags before flags specific to e.g., stargz/nydus/etc